Backend/mentii/class_ctrl.py

```diff
@@ -183,11 +186,144 @@ def getPublicClassList(dynamodb, email=None):
+          MentiiLogging.getLogger().error('Unable to update classes in removeStudentFromClass')
+          response.addError('Failed to remove student from class', 'Unable to update user data')
+        else:
+          response.addToPayload('Class removal success', 'Removed class from student')
+  return response
+
+def removeStudentFromClass(dynamoDBInstance, response, email, classCode):
+  classesTable = dbUtils.getTable('classes', dynamoDBInstance)
+  if classesTable is None:
+    MentiiLogging.getLogger().error('Unable to get classes table in removeStudentFromClass')
+    response.addError('Remove Student From Class Failed.', 'Unable to locate necessary table(s).')
+  else:
+    # check that class exists and class has that user
+    cl = getClassByCode(classCode, dynamoDBInstance)
+    if cl is None:
+      MentiiLogging.getLogger().error('Unable to get class by classCode in removeStudentFromClass')
+      response.addError('Failed to remove student from class', 'Unable to get class')
+    else:
+      students = cl['students'] #set
+      if email not in students:
+        response.addError('Failed to remove student from class', 'Student not found')
+      else:
+        students.remove(email)
+        jsonData = {}
+        if len(students) == 0:
+          #remove attribute
+          jsonData = {
+            'Key': {'code': classCode},
+            'UpdateExpression': 'REMOVE students',
+            'ReturnValues' : 'UPDATED_NEW'
+          }
+        else:
+          #update attribute
+          jsonData = {
+            'Key': {'code': classCode},
+            'UpdateExpression': 'SET students = :s',
+            'ExpressionAttributeValues': { ':s': students },
+            'ReturnValues' : 'UPDATED_NEW'
+          }
+        res = dbUtils.updateItem(jsonData, classesTable)
+        if res is None:
+          MentiiLogging.getLogger().error('Unable to update classes in removeStudentFromClass')
+          response.addError('Failed to remove student from class', 'Unable to update classes data')
+        if response.hasErrors():
+          # add classCode back to student if there was an error removing student from class
+          usersTable = dbUtils.getTable('users', dynamoDBInstance)
+          user = user_ctrl.getUserByEmail(email, dynamoDBInstance)
+          jsonData = {}
+          if user.get('classCodes') is None:
+            user_ctrl.addClassCodeToStudent(email, classCode, dynamoDBInstance)
+          else:
+            classes = user.get('classCodes') #set
+            classes.add(classCode)
+            #update item
+            jsonData = {
+              'Key': {'email': email},
+              'UpdateExpression': 'SET classCodes = :cc',
+              'ExpressionAttributeValues': { ':cc': classes },
+              'ReturnValues' : 'UPDATED_NEW'
+            }
+          dbUtils.updateItem(jsonData, usersTable)
+        else:
+          response.addToPayload('Student removal success', 'Removed student from class')
```

Comment:
Similar thing with this payload add. We could add a line to the log for success if you really want to put it somewhere.